Engine: Improve exitcode detection and logging

euank opened the pull request:
Prior to this change, the 'exitcode' would be set in any case where the
container was not running, including created. This scopes it down to
only be after it exists. This also improves logging.

In addition, this ensures we log the DockerID and name explicitly at
info level since recovering that mapping from logs is often valuable

Example of the log lines showing name and ID:
```
2015-11-25T23:04:09Z [INFO] Created container name mapping for task sleep5:3 arn:aws:ecs:us-west-2:123:task/my-arn-uuid, Status: (NONE->RUNNING) Containers: [sleep5 (PULLED->RUNNING),] - sleep5(busybox) (PULLED->RUNNING) -> ecs-sleep5-3-sleep5-beb0a7b5ef9683ac6f00
2015-11-25T23:04:09Z [INFO] Created docker container for task sleep5:3 arn:aws:ecs:us-west-2:123:task/my-arn-uuid, Status: (NONE->RUNNING) Containers: [sleep5 (PULLED->RUNNING),]: sleep5(busybox) (PULLED->RUNNING) -> 2d3f2239d8f027cd244ca4f484872217b9edfed032b849828d80c3ae2d6b88e6
2015-11-25T23:04:09Z [INFO] Redundant container state change for task sleep5:3 arn:aws:ecs:us-west-2:123:task/my-arn-uuid, Status: (CREATED->RUNNING) Containers: [sleep5 (CREATED->RUNNING),]: sleep5(busybox) (CREATED->RUNNING) to CREATED, but already CREATED
```

I also verified functional tests are still passing with this.

r? @samuelkarp, @juanrhenals

aaithal commented on agent/api/status.go:
```diff
@@ -100,9 +100,6 @@ func (cs *ContainerStatus) Terminal() bool {
 	return *cs == ContainerStopped
```
cs == ContainerStopped?

euank commented:
I fixed the test I missed before and made the improvement you suggested @aaithal. I also updated the existing create test to reject the wrong exit code behavior.

samuelkarp commented:
:+1:
